demo/mkldnn_quant/quant_aware/PaddleCV_mkldnn_quantaware_tutorial.md

```diff
@@ -0,0 +1,220 @@
+
+## 3. Convert fp32 qat model to MKL-DNN INT8 model
+The model saved after training in the previous step is the float32 qat model. We also need to remove the quantization, inverse quantization op, fuse some ops, and fully converted into INT8 model. Run the following script
+
+```
+python ./save_qat_model.py --qat_model_path=$PATH_TO_float32_qat_dir --int8_model_save_path=$PATH_TO_SAVE_INT8_MODEL --quantized_ops="conv2d,pool2d"
+```
+
+## 4. Inference test
+
+### 4.1 Data preprocessing
+To run the inference test, the data needs to be converted to binary first. Run the script without any pramaters to transform the complete ILSVRC2012 val data set to bianary file. Use `local` parameter to transform your own data.
+```
+python ../tools/full_ILSVRC2012_val_preprocess.py --local --data_dir=$USER_DATASET_PATH --output_file=data.bin  
+```
+
+Optional parameters:
+- No parameters set. The script will download the ILSVRC2012_img_val data set and convert it into a binary file.
+- local, set to true, indicating that users will provide their own data
+- data_dir, set the user's own data directory
+- label_list, set the image path-image category list, similar to `val_list.txt`
+- output_file, the name of the generated bin file
+- data_dim, the length and width of the preprocessed image, the default is 224. Not recommended to change
+
+The user's own data set directory structure should be as follows
+```
+imagenet_user
+├── val
+│ ├── ILSVRC2012_val_00000001.jpg
+│ ├── ILSVRC2012_val_00000002.jpg
+| | ── ...
+└── val_list.txt
+```
+Among them, the content of val_list.txt should be as follows:
+```
+val / ILSVRC2012_val_00000001.jpg 0
+val / ILSVRC2012_val_00000002.jpg 0
+```
+
+note:
+- Why convert the data set into a binary file? Because the data preprocessing (resize, crop, etc.) in Paddle is performed using pythong.Image module, the trained model is also based on the images preprocessed by Python. However, we found python test has high performance overhead and the inference performance slow down. Hence, in the quantitative model inference stage, we decided to use C++ tests. While C ++ only supports libraries such as Open-CV, paddle does not recommend the use of external libraries, so we use Python to preprocess the images and put data into a binary file. Then read it into C++ test. Users can also change the C++ test to use open-cv library to directly read the data and preprocess it, and the accuracy will not be greatly reduced. We also provide a python test `sample_tester.py` as a reference, users can see its high performance overhead compared to C++ test `sample_tester.cc`.
```

Comment:
Maybe: "The reason for converting the dataset into a binary file is performance and independence of external C++ libraries. Image data requires preprocessing like resizing, cropping, etc. and it can be easily achieved using python Image module, both for training and inference. However, the performance of python tests is lower than of C++ tests, hence the decision to latter for the quantitative model inference tests. While effective image processing in C++ requires linking to external libraries like Open-CV, to avoid adding new dependencies to Paddle we use Python to preprocess the image data and put the result in a binary file. Then, the binary dataset is read into the C++ test. A python test `sample_tester.py` is there for reference, so users can observe its performance overhead compared to the C++ test `sample_tester.cc`.